src/nft/components/collection/CollectionStats.tsx

```diff
@@ -0,0 +1,328 @@
+            collectionSocialsIsOpen={collectionSocialsIsOpen}
+            toggleCollectionSocials={toggleCollectionSocials}
+          />
+        )}
+    </Row>
+  )
+}
+
+const CollectionDescription = ({ description }: { description: string }) => {
+  const [showReadMore, setShowReadMore] = useState(false)
+  const [readMore, toggleReadMore] = useReducer((state) => !state, false)
+  const baseRef = useRef<HTMLDivElement>(null)
+  const descriptionRef = useRef<HTMLDivElement>(null)
+
+  useEffect(() => {
+    if (
+      baseRef &&
+      descriptionRef &&
+      baseRef.current &&
+      descriptionRef.current &&
+      (descriptionRef.current.getBoundingClientRect().width >= baseRef.current?.getBoundingClientRect().width - 112 ||
+        descriptionRef.current.getBoundingClientRect().width >= 590)
+    )
+      setShowReadMore(true)
+  }, [descriptionRef, baseRef])
+
+  return (
+    <Box ref={baseRef} marginTop={{ mobile: '12', tabletSm: '16' }} style={{ maxWidth: '680px' }}>
+      <Box
+        ref={descriptionRef}
+        className={clsx(styles.description, styles.nameText, readMore && styles.descriptionOpen)}
+      >
+        <ReactMarkdown
+          source={description}
+          allowedTypes={['link', 'paragraph', 'strong', 'code', 'emphasis', 'text']}
+          renderers={{ paragraph: 'span' }}
+        />
+      </Box>
+      <Box as="span" display={showReadMore ? 'inline' : 'none'} className={styles.readMore} onClick={toggleReadMore}>
+        Show {readMore ? 'less' : 'more'}
+      </Box>
+    </Box>
+  )
+}
+
+const StatsItem = ({ children, label, isMobile }: { children: ReactNode; label: string; isMobile: boolean }) => (
+  <Box display="flex" flexDirection={isMobile ? 'row' : 'column'} alignItems="baseline" gap="2" height="min">
+    <Box as="span" className={styles.statsLabel}>
+      {`${label}${isMobile ? ': ' : ''}`}
+    </Box>
+    <span className={styles.statsValue}>{children}</span>
+  </Box>
+)
+
+const StatsRow = ({ stats, isMobile, ...props }: { stats: GenieCollection; isMobile?: boolean } & BoxProps) => {
+  const numOwnersStr = stats.stats ? putCommas(stats.stats.num_owners) : 0
+  const totalSupplyStr = stats.stats ? putCommas(stats.stats.total_supply) : 0
+  const totalListingsStr = stats.stats ? putCommas(stats.stats.total_listings) : 0
+
+  // round daily volume & floorPrice to 3 decimals or less
+  const totalVolumeStr = ethNumberStandardFormatter(stats.stats?.total_volume)
+  const floorPriceStr = ethNumberStandardFormatter(stats.floorPrice)
+
+  return (
+    <Row gap={{ mobile: '20', tabletSm: '60' }} {...props}>
+      <StatsItem label="Items" isMobile={isMobile ?? false}>
+        {totalSupplyStr}
+      </StatsItem>
+      {numOwnersStr ? (
+        <StatsItem label="Owners" isMobile={isMobile ?? false}>
+          {numOwnersStr}
+        </StatsItem>
+      ) : null}
+      {stats.floorPrice ? (
+        <StatsItem label="Floor Price" isMobile={isMobile ?? false}>
+          {floorPriceStr} ETH
```

Comment:
Is this only meant for mainnet (with ETH hardcoded), or should it be updated to reflect the native token for any chain?

Reply:
currently the specs are that we are only supporting NFTs on mainnet